xarray/backends/zarr.py

```diff
@@ -212,7 +212,13 @@ def extract_zarr_variable_encoding(
     """
     encoding = variable.encoding.copy()
 
-    valid_encodings = {"chunks", "compressor", "filters", "cache_metadata"}
+    valid_encodings = {
```

Comment:
Do we really need this check at all? Does Zarr not check for invalid encodings?

Reply:
Thanks for taking a look @dcherian.

Zarr will issue a UserWarning for invalid encodings (kwargs):

```python
>>> zarr.create((1,), blah=1)
/Users/tom/miniconda/envs/sgkit-3.8/lib/python3.8/site-packages/zarr/creation.py:221: UserWarning: ignoring keyword argument 'blah'
  warn('ignoring keyword argument %r' % k)
<zarr.core.Array (1,) float64>
```

However, it looks like `extract_zarr_variable_encoding` is using the list of valid encodings to find the relevant kwargs to pass to Zarr, so this list may well be needed in order to not trigger the warning if a variable has other, non-Zarr encodings.

Reply:
Hmm... I'll merge and lets discuss in an issue.